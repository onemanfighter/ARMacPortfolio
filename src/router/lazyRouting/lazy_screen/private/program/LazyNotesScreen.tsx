import { lazy } from 'react';
import { LazyProvider } from '../../../component';

const Notes = lazy(
  () => import('../../../../../screens/private/program/Notes/Notes'),
);

const LazyNotesComponent = () => {
  return (
    <LazyProvider>
      <Notes />
    </LazyProvider>
  );
};

export { LazyNotesComponent };
