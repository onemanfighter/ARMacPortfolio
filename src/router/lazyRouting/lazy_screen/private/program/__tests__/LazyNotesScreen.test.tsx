import { render, screen, waitFor } from '@testing-library/react';
import { LazyNotesComponent } from '../LazyNotesScreen';
import { renderHook } from '@testing-library/react-hooks';
import { appStore } from '@appStore';
import { processStore } from '@processStore';
import { settingsStore } from '@settingsStore';

describe('LazyNotesComponent', () => {
  it('should render correctly to match snapshot', async () => {
    renderHook(() => appStore());
    renderHook(() => processStore());
    renderHook(() => settingsStore());

    const { container } = render(<LazyNotesComponent />);

    await waitFor(() =>
      expect(screen.getByLabelText('notes-app')).toBeDefined(),
    );
    expect(container).toMatchSnapshot();
  });
});
