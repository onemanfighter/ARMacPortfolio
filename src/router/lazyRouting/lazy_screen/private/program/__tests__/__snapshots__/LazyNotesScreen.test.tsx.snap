// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyNotesComponent should render correctly to match snapshot 1`] = `
<div>
  <div
    aria-label="notes-app"
    class="css-1v8gcgw"
  >
    <div
      class="css-cum3m2"
    >
      <button
        aria-label="add-note"
        class="chakra-button css-p7mkev"
        type="button"
      >
        <p
          aria-hidden="true"
          class="chakra-text css-bk9fzy"
          focusable="false"
        >
          +
        </p>
      </button>
      <button
        aria-label="delete-note"
        class="chakra-button css-p7mkev"
        type="button"
      >
        <p
          aria-hidden="true"
          class="chakra-text css-bk9fzy"
          focusable="false"
        >
          -
        </p>
      </button>
    </div>
    <div
      class="css-1fjg3y4"
    >
      <div
        class="css-1myg2cv"
      />
      <div
        aria-label="note-editor-box"
        class="css-1xb7lh4"
      >
        <div>
          Editor
        </div>
      </div>
    </div>
  </div>
</div>
`;
